<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>readme</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="README_files/libs/clipboard/clipboard.min.js"></script>
<script src="README_files/libs/quarto-html/quarto.js"></script>
<script src="README_files/libs/quarto-html/popper.min.js"></script>
<script src="README_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="README_files/libs/quarto-html/anchor.min.js"></script>
<link href="README_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="README_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="README_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="README_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="README_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">



<section id="zosterasnps" class="level1">
<h1>ZosteraSNPs</h1>
<p>This is the plant trait and SNP dataset for Zostera marina Updated Dec 30 2023</p>
<section id="map-of-sites-and-geographic-distances" class="level3">
<h3 class="anchored" data-anchor-id="map-of-sites-and-geographic-distances">map of sites and geographic distances</h3>
<p><strong><em>R/SiteMaps.R</em></strong><br>
<strong><em>data/NSDE41661/</em></strong><br>
<strong><em>output/GeographicDistanceIndiv.csv</em></strong><br>
<strong><em>output/IndivLatLon.csv</em></strong><br>
<strong><em>output/SiteQuadratLatLon.csv</em></strong><br>
<strong><em>output/SiteMaps_coastline.pdf</em></strong><br>
<strong><em>output/SiteMaps.pdf</em></strong></p>
</section>
<section id="adult-plant-traits" class="level3">
<h3 class="anchored" data-anchor-id="adult-plant-traits">Adult plant traits</h3>
<p><strong><em>R/DensityHeightAnalysis.R</em></strong><br>
<strong><em>data/perm.quad.density.2019.csv</em></strong><br>
<strong><em>data/perm.quad.veg.ht.2019.csv</em></strong><br>
<strong><em>output/DensityHeight.png</em></strong></p>
</section>
<section id="genotype-likelihoods-and-calls-are-in-datasnps" class="level3">
<h3 class="anchored" data-anchor-id="genotype-likelihoods-and-calls-are-in-datasnps">Genotype likelihoods and calls are in data/SNPs/</h3>
</section>
<section id="analysis-of-clonality-via-ngsrelate" class="level3">
<h3 class="anchored" data-anchor-id="analysis-of-clonality-via-ngsrelate">Analysis of clonality via ngsRelate</h3>
<p><strong><em>data/ngsRelateFiles/</em></strong> The output from ngsRelate<br>
<strong><em>R/ngsRelate_processFiles_rab.R</em></strong><br>
<strong><em>output/histogram_ngsRelate_rab.pdf</em></strong><br>
<strong><em>output/clones_r=70perc.txt</em></strong><br>
<strong><em>output/clones_r=80perc.txt</em></strong></p>
</section>
<section id="analysis-of-clonality-via-dissimilarity" class="level3">
<h3 class="anchored" data-anchor-id="analysis-of-clonality-via-dissimilarity">Analysis of clonality via dissimilarity</h3>
<p><strong><em>data/DissimilarityDist_clones_95perc.csv.gz</em></strong><br>
<strong><em>R/dissimilarity.R</em></strong><br>
<strong><em>output/histogram_dissimilarity.pdf</em></strong></p>
</section>
<section id="finalize-list-of-clones-and-make-map" class="level3">
<h3 class="anchored" data-anchor-id="finalize-list-of-clones-and-make-map">finalize list of clones and make map</h3>
<p><strong><em>R/IdentifyClones_summary.R</em></strong><br>
<strong><em>output/ListOfClones.txt</em></strong><br>
<strong><em>output/Clone_analysis.csv</em></strong><br>
<strong><em>data/214adults99seeds_noClones_clean</em></strong> The output of unique genets to use<br>
<strong><em>R/ngsRelate_ClonesOnMap.R</em></strong><br>
<strong><em>output/output/ClonesOnMap_Both.pdf</em></strong></p>
</section>
<section id="seed-density-and-size" class="level3">
<h3 class="anchored" data-anchor-id="seed-density-and-size">Seed density and size</h3>
<p><strong><em>data/seedPhenotypes.csv</em></strong><br>
<strong><em>R/seedPhenotypes.R</em></strong><br>
<strong><em>output/seedPhenotypes.png</em></strong></p>
</section>
<section id="map-of-sites-and-admix-results" class="level3">
<h3 class="anchored" data-anchor-id="map-of-sites-and-admix-results">map of sites and admix results</h3>
<p><strong><em>R/SiteMaps_NGSadmix.R</em></strong><br>
<strong><em>data/NSDE41661/</em></strong><br>
<strong><em>data/k05run1.qopt</em></strong><br>
<strong><em>output/SiteMaps_NGSadmix.pdf</em></strong></p>
</section>
<section id="analysis-of-molecular-variance-amova-and-pairwise-phist" class="level3">
<h3 class="anchored" data-anchor-id="analysis-of-molecular-variance-amova-and-pairwise-phist">Analysis of Molecular Variance (AMOVA) and Pairwise PhiST</h3>
<p><strong><em>R/amova_noClones_calls.R</em></strong><br>
<strong><em>R/amova_seeds.R</em></strong><br>
<strong><em>output/amova.method1-calls_noClones.csv</em></strong><br>
<strong><em>output/amova.method1-calls_noClones-pairwisePhiSt.csv</em></strong><br>
<strong><em>output/amova.method1-calls_seeds-pairwisePhiSt.csv</em></strong><br>
<strong><em>output/amova.method1-seeds.csv</em></strong></p>
</section>
<section id="to-do" class="level3">
<h3 class="anchored" data-anchor-id="to-do">~~~~~ TO DO ~~~~~~~~</h3>
</section>
<section id="angsd-generated-genotype-likelihoods-in-beagle-format" class="level3">
<h3 class="anchored" data-anchor-id="angsd-generated-genotype-likelihoods-in-beagle-format">angsd-generated genotype likelihoods in beagle format</h3>
<p><strong><em>data/ind393_clean</em></strong><br>
<strong><em>data/loci19433</em></strong><br>
<strong><em>data/zos.393ind.HWE.99.gl.beagle.gz</em></strong></p>
</section>
<section id="genotype-calls" class="level3">
<h3 class="anchored" data-anchor-id="genotype-calls">genotype calls</h3>
<p><strong><em>R/SubsetGenotypeCalls.R</em></strong><br>
<strong><em>data/zos.393ind.HWE.99.loci19433.vcf.gz</em></strong><br>
<strong><em>data/genotypeCalls.393ind.8684loci.geno</em></strong>This is the output dataset<br>
<strong><em>data/ind393_clean</em></strong><br>
<strong><em>data/loci8684</em></strong></p>
</section>
<section id="spaghedi-correlogram-of-adults" class="level3">
<h3 class="anchored" data-anchor-id="spaghedi-correlogram-of-adults">SPAGHEDI correlogram of adults</h3>
<p><strong><em>R/CreateSpaghediFiles.R</em></strong><br>
<strong><em>R/SpagediSGS_noClones_10categories.R</em></strong><br>
<strong><em>data/allSite.Spagedi.out_noClones_10categories.txt</em></strong><br>
<strong><em>output/SpagediSGS_noClones_10categories.pdf</em></strong></p>
</section>
<section id="basic-pop-gen-statistics" class="level3">
<h3 class="anchored" data-anchor-id="basic-pop-gen-statistics">Basic pop gen statistics</h3>
<p><strong><em>R/basicStats_adultsNoClones.R</em></strong><br>
<strong><em>output/BasicStats_adultsNoClones.csv</em></strong></p>
</section>
<section id="create-fasta-from-calls-and-generate-maximum-likelihood-tree" class="level3">
<h3 class="anchored" data-anchor-id="create-fasta-from-calls-and-generate-maximum-likelihood-tree">Create FASTA from calls and generate Maximum Likelihood tree</h3>
<p><strong><em>R/ConvertVCFToFASTA.R</em></strong><br>
<strong><em>R/iQtreePlot.R</em></strong><br>
<strong><em>data/fasta/XXX</em></strong><br>
<strong><em>data/fasta/XXX</em></strong><br>
<strong><em>data/fasta/XXX</em></strong></p>
</section>
<section id="isolation-by-distance" class="level3">
<h3 class="anchored" data-anchor-id="isolation-by-distance">Isolation by distance</h3>
<p><strong><em>R/Fst~Distance.R</em></strong><br>
<strong><em>output/amova.method1-calls_noClones-pairwisePhiSt.csv</em></strong><br>
<strong><em>output/Fst~Distance.pdf</em></strong></p>
</section>
<section id="admixture-analysis---adults-and-seeds" class="level3">
<h3 class="anchored" data-anchor-id="admixture-analysis---adults-and-seeds">Admixture analysis - adults and seeds</h3>
<p><strong><em>R/ngsAdmix_dips.pretty_adults-NoClones.R</em></strong><br>
<strong><em>R/ngsAdmix_dips.pretty_seeds.R</em></strong><br>
<strong><em>data/ind245adults_noClones_clean</em></strong><br>
<strong><em>data/ind99_seeds</em></strong><br>
<strong><em>data/ngsadmix-noClones/</em>opt</strong>*<br>
<strong><em>data/NGSadmix-seeds/</em>opt</strong>*<br>
<strong><em>output/ngsAdmix_dips.pretty-adults-NoClones.pdf</em></strong><br>
<strong><em>output/ngsAdmix_dips.pretty-seeds.pdf</em></strong></p>
</section>
<section id="site-specific-pca-and-admix-plots" class="level3">
<h3 class="anchored" data-anchor-id="site-specific-pca-and-admix-plots">Site-specific PCA and admix plots</h3>
<p><strong><em>R/Site-pca&amp;admix_noClones.R</em></strong><br>
<strong><em>data/zos.393ind.HWE.99.loci19433.vcf.gz</em></strong><br>
<strong><em>data/ind393_clean</em></strong><br>
<strong><em>data/ind245adults_noClones_clean</em></strong><br>
<strong><em>data/loci8684</em></strong><br>
<strong><em>data/ngsadmix-noClones-site/</em></strong><br>
<strong><em>output/Site-pca&amp;admix_noClones.pdf</em></strong></p>
</section>
<section id="relatedness-of-seeds-within-and-among-grids-and-cores-ngsrelate" class="level3">
<h3 class="anchored" data-anchor-id="relatedness-of-seeds-within-and-among-grids-and-cores-ngsrelate">Relatedness of seeds within and among grids and cores (NGSrelate)</h3>
<p><strong><em>R/Relatedness-seeds-ngsRelate.R</em></strong><br>
<strong><em>data/Seed ID to core.xlsx</em></strong><br>
<strong><em>data/ngsRelate/</em></strong><br>
<strong><em>output/Relatedness~Core-ngsRelate_seeds.pdf</em></strong></p>
</section>
<section id="dapc-assignment---seeds-adults" class="level3">
<h3 class="anchored" data-anchor-id="dapc-assignment---seeds-adults">DAPC assignment - seeds &amp; adults</h3>
<p><strong><em>R/SeedAssignment-DAPC.R</em></strong><br>
<strong><em>data/genotypeCalls.393ind.8684loci.geno</em></strong><br>
<strong><em>output/SeedAssignment-DAPC.txt</em></strong><br>
<strong><em>output/SeedAssignment-DAPC.pdf</em></strong></p>
</section>
</section>
<section id="commands-from-other-programs" class="level1">
<h1>Commands from other programs</h1>
<section id="ngsrelate" class="level3">
<h3 class="anchored" data-anchor-id="ngsrelate">ngsRelate</h3>
<p><strong><em>see https://github.com/ANGSD/NgsRelate</em></strong><br>
First we generate a file with allele frequencies (angsdput.mafs.gz) and a file with genotype likelihoods (angsdput.glf.gz).***<br>
<strong><em>angsd -b DD_43_bamfiles -gl 2 -domajorminor 1 -snp_pval 1e-6 -domaf 1 -minmaf 0.05 -doGlf 3 -rf zos.393ind.HWE.99.forANGSD.loci19433.txt -out DD</em></strong><br>
Then we extract the frequency column from the allele frequency file and remove the header (to make it in the format NgsRelate needs)<br>
<strong><em>zcat DD.mafs.gz | cut -f5 |sed 1d &gt; DD_freq</em></strong><br>
run NgsRelate The output should be a file called newres that contains the output for all pairs<br>
<strong><em>./ngsRelate -g DD.glf.gz -n 43 -f DD_freq -O DD_newres</em></strong></p>
</section>
<section id="bwa-version-0.7.17-and-samtools-version-1.9-to-align-reads-and-create-bamfiles" class="level3">
<h3 class="anchored" data-anchor-id="bwa-version-0.7.17-and-samtools-version-1.9-to-align-reads-and-create-bamfiles">bwa (Version: 0.7.17) and samtools (Version: 1.9) to align reads and create bamfiles</h3>
<p><strong><em>### pull in lines from fastq list and perform bwa</em></strong><br>
<strong><em>#! /bin/bash</em></strong><br>
<strong><em>cat fastqfiles | while read LINE</em></strong><br>
<strong><em>do</em></strong><br>
<strong><em>bwa mem ../genome/zostera_genome_forBWA -t 20 ../fastq/$LINE.fastq | samtools sort - | samtools view -h - | samtools view -b &gt; $LINE.bwa.bam</em></strong><br>
<strong><em>samtools index $LINE.bwa.bam</em></strong><br>
<strong><em>done</em></strong></p>
</section>
<section id="bcftools-for-calling-snps-version-1.9-259" class="level3">
<h3 class="anchored" data-anchor-id="bcftools-for-calling-snps-version-1.9-259">bcftools for calling SNPs (Version: 1.9-259)</h3>
<p><em><strong>bcftools mpileup -d 500 -Ou -a AD -f ../genome/GCA_001185155.1_Zosma_marina.v.2.1_genomic.fna -b ind393 |*</strong><br>
</em><strong>bcftools call -mv -Ou -G Sample_group.txt |*</strong><br>
<strong><em>bcftools view -m2 -M2 -q 0.01:minor -o zos.393ind.HWE.99.vcf</em></strong><br>
<strong><em>bgzip zos.393ind.HWE.99.vcf</em></strong><br>
<strong><em>bcftools index zos.393ind.HWE.99.vcf.gz</em></strong><br>
<strong><em>bcftools query -f ‘%CHROM%POS%POS’ zos.393ind.HWE.99.vcf.gz &gt; zos.393ind.HWE.99.allloci.txt</em></strong></p>
</section>
<section id="creating-angsd-formatted-beagle-file-version-0.931" class="level3">
<h3 class="anchored" data-anchor-id="creating-angsd-formatted-beagle-file-version-0.931">creating angsd-formatted beagle file (version: 0.931)</h3>
<p><strong><em>angsd -GL 1 -out zos.393ind.HWE.99.geno -doGlf 2 -rf zos.393ind.HWE.99.allloci.forANGSD.txt -doMaf 2 -doMajorMinor 1 -SNP_pval 0.000001 -doGeno 5 -doPost 1 -bam ind393</em></strong></p>
</section>
<section id="iqtree-version-1.6.11" class="level3">
<h3 class="anchored" data-anchor-id="iqtree-version-1.6.11">iQTree (version 1.6.11)</h3>
<p>***Command: /usr/local/iqtree/bin/iqtree -s /tmp/IQTREE/_aGIwYTklJ/seqin.fst -st DNA -pre /tmp/IQTREE/_aGIwYTklJ/iqt -m GTR+F -bb 1000 -pers 0.5 -numstop 100 -nt 3***</p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>